Assets/Shared/OtspIII Simple Games/Breakout/BallController.cs

```diff
@@ -0,0 +1,82 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace OtspIII.SimpleGames
+{
+    namespace Breakout
+    {
+        public class BallController : MonoBehaviour
+        {
+            //My Rigibody
+            public Rigidbody2D RB;
+            //My starting velocity. This should be set in the editor
+            public Vector2 StartVel;
+            //My starting position, where I respawn into. I set this in Start()
+            public Vector3 StartPos;
+            
+            void Start()
+            {
+                //I record where I started, so I can respawn there
+                StartPos = transform.position;
+                //I check my StartVelocity variable and set that to be my velocity
+                RB.linearVelocity = StartVel;
+            }
+
+            void Update()
+            {
+                //If I'm off-screen, I respawn with my initial position & speed
+                if (transform.position.y < -10)
+                {
+                    transform.position = StartPos;
+                    RB.linearVelocity = StartVel;
+                }
+            }
+
+            private void OnTriggerEnter2D(Collider2D other)
+            {
+                //If I hit something, I'm going to bounce. Let's calculate my new velocity
+                Vector2 vel = RB.linearVelocity;
+                
+                //Did I hit the paddle?
+                PaddleController pc = other.gameObject.GetComponent<PaddleController>();
+                if (pc != null)
+                { 
+                    //If so, I should bounce back up
+                    vel.y *= -1;
+                    //I should also be aimed based on where I hit the paddle
+                    //I ask the paddle to calculate this for me
+                    vel.x = pc.BounceAngle(this);
+                }
+
+                //Did I hit a brick?
+                BrickController bc = other.gameObject.GetComponent<BrickController>();
+                if (bc != null)
+                {
+                    //If so, I bounce vertically
+                    //MINOR BUG: if I hit a brick from the side I should bounce horizontally
+                    vel.y *= -1;
+                    //Also I tell the brick to break
+                    bc.Break();
+                }
```

Comment:
Please change the variable to something more descriptive like `brickController`.

Reply:
NO